separated cat and telegram services

evilbream opened the pull request.

evilbream committed:
separated cat and telegram services

shdionis commented on cat_service/src/main/java/com/baranova/cat_service/CatServiceApplication.java:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@ComponentScan(basePackages = "com.baranova.cat_service")
```
а зачем?

shdionis commented on cat_service/src/main/java/com/baranova/cat_service/rabbitMQ/rabbitMQConsumer.java:
```diff
@@ -0,0 +1,36 @@
+
+@Slf4j
+@Component
+public class rabbitMQConsumer {
```
С Большой буквы классы

shdionis commented on cat_service/src/main/java/com/baranova/cat_service/rabbitMQ/config/rabbitMQConsumerConfig.java:
```diff
@@ -0,0 +1,38 @@
+
+@Configuration
+@EnableRabbit
+public class rabbitMQConsumerConfig {
```
и тут

shdionis commented on cat_service/src/main/resources/application.yaml:
```diff
@@ -25,6 +26,27 @@ spring:
+      key_get: key_get
+      key_send: key_send
+
+    #listener:
```
?

shdionis commented on cat_service/src/main/java/com/baranova/cat_service/commands/CommandAddCatName.java:
```diff
@@ -1,25 +1,29 @@
+import com.baranova.cat_service.constants.MessageCallback;
+import com.baranova.cat_service.constants.UserMessage;
 
 public class CommandAddCatName extends AbsCommand {
```
простой Ctrl+C Ctrl+V - это не разделение и так не пойдет. Этот свервис не должен знать вообще ничего о специфике телеграма, командах, сообщениях, меню-кнопках и т.п. Телеграм - одна из реализаций взаимодействия с этим сервисом, но не единственная. В сервисе должен быть реализован CRUD для объектов котиков/пользователей, а телеграм-бот уже сам на основании своих внутренних сущностей должен превращать их в запросы в этот сервис. 
Считайте, что помимо телеграма на проекте еще есть мобильное приложение, сайт, десктоп-приложение и viber-бот. Этот сервис должен общаться с ними одинаково, а уже на самих клиентах будет превращение взаимодействия с этим сервисом во взаимодействие с пользователем

evilbream committed:
+

evilbream committed:
1. Исправлено перемешение кнопок, теперь кнопки располагаются в одном порядке
2. Исправлена ошибка записи в бд при отправке реакции
3. Исправлена ошибка пагинации

evilbream committed:
+

evilbream committed:
добавлен shared модуль

evilbream committed:
cat service removed commands
